genie-agent/src/main/java/com/netflix/genie/agent/execution/services/LaunchJobServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+        }
+    }
+
+    private void configureEnvironment(
+        final ProcessBuilder processBuilder
+    ) throws JobLaunchException {
+
+        final File jobDirectory = executionContext.getJobDirectory();
+
+        // Validate job running directory
+        if (jobDirectory == null) {
+            throw new JobLaunchException("Job directory is not set in execution context");
+        } else if (!jobDirectory.exists()) {
+            throw new JobLaunchException("Job directory does not exist: " + jobDirectory);
+        } else if (!jobDirectory.isDirectory()) {
+            throw new JobLaunchException("Job directory is not a directory: " + jobDirectory);
+        } else if (!jobDirectory.canWrite()) {
+            throw new JobLaunchException("Job directory is not writeable: " + jobDirectory);
+        }
+
+        // Configure job running directory
+        processBuilder.directory(jobDirectory);
+
+        final Map<String, String> currentEnvironmentVariables = processBuilder.environment();
+        final Map<String, String> jobEnvironmentVariables = executionContext.getJobEnvironment();
+
+        // Validate additional environment variables
+        if (jobEnvironmentVariables == null) {
+            throw new JobLaunchException("Job environment variables map is not set in execution context");
+        }
+
+        // Configure job environment variables
+        jobEnvironmentVariables.forEach((key, value) -> {
+            final String replacedValue = currentEnvironmentVariables.put(key, value);
+            if (StringUtils.isBlank(replacedValue)) {
+                log.debug(
+                    "Added job environment variable: {}={}",
+                    key,
+                    value
+                );
+            } else if (!replacedValue.equals(value)) {
+                log.debug(
+                    "Set job environment variable: {}={} (previous value: {})",
+                    key,
+                    value,
+                    replacedValue
+                );
+            }
+        });
+    }
+
+    private void configureCommandArguments(
+        final ProcessBuilder processBuilder,
+        final JobSpecification jobSpec
+    ) throws JobLaunchException {
+        final List<String> commandLineArguments = jobSpec.getCommandArgs();
+
+        // Validate arguments
+        if (commandLineArguments == null) {
+            throw new JobLaunchException("Job command-line arguments not set in execution context");
+        } else if (commandLineArguments.isEmpty()) {
+            throw new JobLaunchException("Job command-line arguments are empty");
+        }
+
+        // Configure arguments
+        processBuilder.command(commandLineArguments);
+    }
+
+    private void configureIO(
+        final ProcessBuilder processBuilder,
+        final JobSpecification jobSpec
+    ) {
+        if (jobSpec.isInteractive()) {
+            processBuilder.inheritIO();
```

Comment:
Have we double checked with @danielcweeks and @rdblue  that this is the behavior they want? If I'm not mistaken now there won't be any data in stdout and stderr when a job is interactive.

Reply:
Inherit means that it's inherited from the terminal, so if a command prints 'foo', the user sees 'foo' in the terminal. 
If the user writes 'bar', it will be forwarded to the process.

It is the simplest behavior for interactive commands.

If we do also want to intercept in/out/err and log it to a file, then we need to intercept this and it'll be more complex.
But in terms of M1, to reach parity with the Runner, I'm fairly sure this is the right setting.

Reply:
Sounds good. We can address it if need be after M1